### PR TITLE
improvement: ignore code-sandbox references in code snipets, add try-…

### PR DESCRIPTION
**Pull request by SofiaAfendra**
…catch statements

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -7,48 +7,63 @@ const slugify = require('./build-utils/slugify/slugify');
...
-      if (sandboxPath) {
-        sandboxPath = `${__dirname}/__sandboxes__/_examples/${sandboxPath}`;
-        const createdSandboxId = await generateSandbox(sandboxPath);
-
-        if (createdSandboxId) {
-          node.context.body = node.context.body.replace(
-            l,
-            l.split('codesandbox-link://')[0] + getSandboxUrl(createdSandboxId) + ')'
-          );
+  const sandboxLinks = node?.context?.body
```
Comment (xristos1994) [inline gatsby-node.js:13]: With he two following `replace` you assume that there is a new line (\n) after every opening backticks and before closing backticks. But, in markdown the lines \`some code\` and \`\`\`some code\`\`\` are totally valid as code blocks. Additionally double backtics are also considered as codeblock.
 So, wha tI would suggest is to use two even simpler regex.
 ```javascript
node?.context?.body
   ?.replace(/``[^`]*``/g, '') // remove everything that is contained inside double backtics including backticks themeselves.
   ?.replace(/`[^`]*`/g, '') // remove everything that is contained inside single backticks including backticks themselves. Note that 3 backticks are included in this case.
   ?.match(/\[.*\]\(.*codesandbox-link:\/\/.*\)/g);
```

File: gatsby-node.js [hunk 2]
```diff
@@ -7,48 +7,63 @@ const slugify = require('./build-utils/slugify/slugify');
...
-      }
-    });
+      });
+    }
+  } catch (error) {
+    console.error(error);
   }
 
-  const sandboxIFrames = node?.context?.body?.match(/\[?.*\]\(?.*codesandbox-embed:\/\/?.*\)/g);
+  const sandboxIFrames = node?.context?.body
```
Comment (xristos1994) [inline gatsby-node.js:40]: The same as the above comment about regex.

**Review by xristos1994 — CHANGES_REQUESTED** [review]

File: gatsby-node.js (hunk 1 above)
Comment (SofiaAfendra) [reply gatsby-node.js:13]: Sure, done!

File: gatsby-node.js (hunk 2 above)
Comment (SofiaAfendra) [reply gatsby-node.js:39]: Done!

**Review by xristos1994 — APPROVED** [review]

**Pull request merged**